src/http-tequilapi-client.ts

```diff
@@ -0,0 +1,155 @@
+import { ConsumerLocationDTO, parseConsumerLocationDTO } from './dto/consumer-location'
+import { parseIdentitiesResponseDTO } from './dto/identities-response'
+import { IdentityDTO, parseIdentityDTO } from './dto/identity'
+import {
+  IdentityRegistrationDTO,
+  parseIdentityRegistrationDTO
+} from './dto/identity-registration/identity-registration'
+import { NodeHealthcheckDTO, parseHealthcheckResponse } from './dto/node-healthcheck'
+import { ProposalDTO } from './dto/proposal'
+import { parseProposalsResponseDTO } from './dto/proposals-response'
+import { ConnectionRequest } from './dto/query/connection-request'
+import { ProposalQueryOptions } from './dto/query/proposals-query-options'
+import { TIMEOUT_DISABLED } from './timeouts'
+
+export class HttpTequilapiClient implements TequilapiClient {
+  public http: HttpInterface
+
+  constructor (http: HttpInterface) {
+    this.http = http
+  }
+
+  public async healthCheck (timeout?: number): Promise<NodeHealthcheckDTO> {
+    const response = await this.http.get('healthcheck', undefined, timeout)
+    if (!response) {
+      throw new Error('Healthcheck response body is missing')
+    }
+    return parseHealthcheckResponse(response)
+  }
+
+  public async stop (): Promise<void> {
+    await this.http.post('stop')
+  }
+
+  public async identitiesList (): Promise<IdentityDTO[]> {
+    const response = await this.http.get('identities')
+    if (!response) {
+      throw new Error('Identities response body is missing')
+    }
+    const responseDto = parseIdentitiesResponseDTO(response)
+
+    return responseDto.identities
+  }
+
+  public async identityCreate (passphrase: string): Promise<IdentityDTO> {
+    const response = await this.http.post('identities', { passphrase })
+    if (!response) {
+      throw new Error('Identities creation response body is missing')
+    }
+    return parseIdentityDTO(response)
+  }
+
+  public async identityUnlock (id: string, passphrase: string, timeout?: number): Promise<void> {
+    await this.http.put('identities/' + id + '/unlock', { passphrase }, timeout)
+  }
+
+  public async identityRegistration (id: string): Promise<IdentityRegistrationDTO> {
+    const response = await this.http.get(`identities/${id}/registration`)
+    if (!response) {
+      throw new Error('Identities registration response body is missing')
+    }
+    return parseIdentityRegistrationDTO(response)
+  }
+
+  public async findProposals (options?: ProposalQueryOptions): Promise<ProposalDTO[]> {
+    const params = options ? new ProposalsQuery(options).toQueryParams() : undefined
+    const response = await this.http.get('proposals', params)
+    if (!response) {
+      throw new Error('Proposals response body is missing')
+    }
+    const responseDto = parseProposalsResponseDTO(response)
+    const proposals = responseDto.proposals
+
+    if (!proposals) {
+      return []
+    }
+    return proposals
```

Comment:
back in my day, this could be written as `return proposals || []`. Not sure if it works in typescript though.